config/mongo.js

```diff
@@ -1,36 +1,29 @@
 import mongoose from "mongoose";
-import * as fs from "fs";
+import { env } from "process";
+import logger from "./logger";
 
-import logger from "../config/logger";
+let connection;
 
-let hasConnection;
 const connectMongo = async () => {
-  if (!process.env.LINKFREE_MONGO_CONNECTION_STRING) {
+  const { LINKFREE_MONGO_CONNECTION_STRING } = process.env;
+
+  if (!LINKFREE_MONGO_CONNECTION_STRING) {
     throw new Error(
       "Please define the LINKFREE_MONGO_CONNECTION_STRING environment variable (if local add to .env file)"
     );
   }
 
-  if (hasConnection) {
-    return hasConnection;
+  if (connection) {
```

Comment:
This check to see if the connection exists should be at the start of the function. If a connection exists, the function can return from there itself.

Reply:
Yeah. It's true. We don't need to check connection strings if there is existing connection available